### PR TITLE
Delete coupons instead of trashing them

### LINKED ISSUE #30371: Cleanup orders and coupons left in the trash after the tests run
We need to cleanup the coupons and orders that have been left in the trash after the tests complete.

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR updates the new coupon test to delete the created coupon instead of trashing it. It adds a `waitForTimeout` utility function and `AdminEdit`, `OrderEdit` utility classes.

Closes #30371 .

### How to test the changes in this Pull Request:

1. Verify CI
2. Run E2E tests locally

### CODE REVIEW

File: tests/e2e/utils/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Unreleased
 
+## Added
+
+- `utils.waitForTimeout( delay )` pause processing for `delay` milliseconds
+- `AdminEdit`, `OrderEdit` classes with utility functions for the respctive edit screens
```
Comment (zhongruige) [inline tests/e2e/utils/CHANGELOG.md]: Minor typo:

```suggestion
- `AdminEdit`, `OrderEdit` classes with utility functions for the respective edit screens
```

File: tests/e2e/utils/README.md [hunk 2]
```diff
@@ -153,6 +153,34 @@ This package provides support for enabling retries in tests:
 | `updatePaymentGateway`| `paymentGatewayId`, `payload` | Update a payment gateway |
 | `getSystemEnvironment` | | Get the current environment from the WooCommerce system status API.
 
+### Classes
+
+The package includes the following page specific utility classes
```
Comment (zhongruige) [inline tests/e2e/utils/README.md]: Just a minor suggestion to add a colon at the end:

```suggestion
The package includes the following page specific utility classes:
```

File: tests/e2e/utils/src/pages/admin-edit.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	 * @returns {Promise<void>}
+	 */
+	async verifyPublish( button, publishNotice, publishVerification ) {
+		await super.verifyPublish( button, publishNotice, publishVerification );
+
+		await expect( page ).toMatchElement( '#select2-order_status-container', { text: 'Processing' } );
+		await expect( page ).toMatchElement(
+			'#woocommerce-order-notes .note_content',
+			{
+				text: 'Order status changed from Pending payment to Processing.',
```
Comment (zhongruige) [inline tests/e2e/utils/src/pages/admin-edit.js]: Do we always expect the `Processing` status and note status text to be present (e.g., `Order status changed from Pending payment to Processing.` ) when using `verifyPublish()` here, or do we need to make them more dynamic?

**Review by zhongruige — CHANGES_REQUESTED** [review]
Thanks for working on this @rrennick! These ran great for me locally.

Only a few minor comments on text changes and a question. Otherwise this looks good to merge once those are addressed.

File: tests/e2e/utils/CHANGELOG.md (hunk 1 above)
Comment (rrennick) [reply tests/e2e/utils/CHANGELOG.md]: Nice catch, fixed in https://github.com/woocommerce/woocommerce/pull/30696/commits/5ccf3a564fb4cc4cb4c392584320c6e6ec20069d

File: tests/e2e/utils/README.md (hunk 2 above)
Comment (rrennick) [reply tests/e2e/utils/README.md]: Fixed in https://github.com/woocommerce/woocommerce/pull/30696/commits/5ccf3a564fb4cc4cb4c392584320c6e6ec20069d

File: tests/e2e/utils/src/pages/admin-edit.js (hunk 3 above)
Comment (rrennick) [reply tests/e2e/utils/src/pages/admin-edit.js]: @zhongruige Thanks for pointing this out. That test is specific to our test flow so I moved that test to there. That eliminated the need for the `OrderEdit` class so I removed it.

This is ready for another review.

**Review by zhongruige — APPROVED** [review]
Thanks for working on this @rrennick!

The changes look good and the tests ran great for me locally.

:shipit:

**Pull request merged**